### PR TITLE
dispose of backgrounded terminals `onBeforeShutdown`

### LINKED ISSUE #7642: Leaking SSH processes with useLocalServer=false
- Set `"remote.SSH.useLocalServer": false`
- Open a Remote-SSH window
- Note `ps ax | grep ssh | wc -l`
- Reload the window
- `ps ax | grep ssh | wc -l` again, it has increased by 2
- Keep reloading the window, more ssh processes accumulate

I assume this is due to terminal persistence for Remote-SSH's local terminals. Since they are background terminals, they shouldn't be kept alive. I think we've fixed this before. If Remote-SSH should be killing its own terminal to prevent this, we can probably do that too.

### PR DESCRIPTION
**Pull request by meganrogge**
fix https://github.com/microsoft/vscode-remote-release/issues/7642

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 1]
```diff
@@ -540,6 +540,11 @@ export class TerminalService implements ITerminalService {
 	}
 
 	private async _onBeforeShutdownAsync(reason: ShutdownReason): Promise<boolean> {
+		if (this._backgroundedTerminalInstances.length) {
+			for (const terminal of this._backgroundedTerminalInstances.filter(b => !b.shouldPersist)) {
+				terminal.dispose();
```
Comment (roblourens) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:545]: Just wondering whether this should be inside the try/catch?

**Review by roblourens — APPROVED** [review]
Thanks!

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:545]: `dispose` won't throw so I think this is fine

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:545]: https://github.com/markw65/vscode/blob/05dce4918275b5e568a52c8c4e318f36dde0bf1f/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1200-L1205

**Comment by meganrogge** [comment]
fyi @Tyriar see https://github.com/microsoft/vscode-remote-release/issues/7642#issuecomment-1532261303

**Pull request merged**